src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -1271,6 +1344,31 @@ void ImageReplayer<I>::handle_stop(int r, Context *on_start, Context *on_stop) {
       return;
     }
 
+    if (m_stopping_for_resync) {
+      m_image_deleter->schedule_image_delete(m_local_pool_id,
+                                             m_local_image_id,
+                                             m_local_image_name,
+                                             m_global_image_id);
+      m_stopping_for_resync = false;
+
+      wait_for_image_delete = true;
+      FunctionContext *ctx = new FunctionContext(
```

Comment:
The ```FunctionContext``` can probably just be a re-invoking of ```handle_stop``` similar to the status update finalizer